src/vma/dev/ring.cpp

```diff
@@ -35,4 +35,21 @@
 
 ring::ring(int count, uint32_t mtu) : m_n_num_resources(count), m_p_n_rx_channel_fds(NULL), m_parent(NULL), m_mtu(mtu)
 {
+#ifdef DEFINED_VMAPOLL
+	m_vma_active = true; /* TODO: This VMA version supports vma_poll() usage mode only */
+	INIT_LIST_HEAD(&m_ec_list);
+	m_vma_poll_completion = NULL;
+#endif // DEFINED_VMAPOLL	
+}
+
+ring::~ring()
+{
+#ifdef DEFINED_VMAPOLL
+	struct ring_ec *ec = get_ec();
+
+	while (ec) {
+		clear_ec(ec);
+		ec = get_ec();
+	}
```

Comment:
There is https://github.com/Mellanox/libvma/pull/253 that is not part of this change.